### PR TITLE
translatesfx: refactor code

### PR DESCRIPTION
**Pull request by pmcollins**

### CODE REVIEW

File: cmd/translatesfx/translatesfx/cli.go [hunk 1]
```diff
@@ -31,9 +31,9 @@ func CLI(args []string) {
 	fmt.Print(translateConfig(paths(args)))
 }
 
-// paths returns two arguments from the command line args:
-// cfgFname is the path to the Smart Agent config that is being translated
-// wd is the working directory used to evaluate file paths found in the Smart Agent config
+// paths returns two arguments from the command line args: cfgFname is the path
+// to the Smart Agent config that is being translated wd is the working
```
Comment (rmfitzpatrick) [inline cmd/translatesfx/translatesfx/cli.go:35]:
```suggestion
// to the Smart Agent config that is being translated and wd is the working
```

File: cmd/translatesfx/translatesfx/testdata/otel-e2e-expected.yaml [hunk 2]
```diff
@@ -59,6 +59,13 @@ service:
         - resourcedetection
       exporters:
         - signalfx
+    metrics:
```
Comment (rmfitzpatrick) [inline cmd/translatesfx/translatesfx/testdata/otel-e2e-expected.yaml:62]: Why would this now be expected (w/ duplicate exporters)?

File: cmd/translatesfx/translatesfx/otel.go [hunk 3]
```diff
@@ -19,104 +19,175 @@ import (
...
+		Receivers: map[string]map[string]interface{}{},
+		Processors: map[string]map[string]interface{}{
+			resourceDetection: {
+				"detectors": []string{"system", "env", "gce", "ecs", "ec2", "azure"},
+			},
+		},
+		Extensions: map[string]map[string]interface{}{},
+		Service: service{
+			Pipelines: map[string]*rpe{
+				"metrics": {
```
Comment (rmfitzpatrick) [inline cmd/translatesfx/translatesfx/otel.go:54]: Not required w/ refactoring but default agent config has forwarder as a metric receiver as well https://github.com/signalfx/splunk-otel-collector/blob/8c4639181c40e0454a07a41a30bd11315b2bede9/cmd/otelcol/config/collector/agent_config.yaml#L164

File: cmd/translatesfx/translatesfx/testdata/otel-e2e-expected.yaml (hunk 2 above)
Comment (pmcollins) [reply cmd/translatesfx/translatesfx/testdata/otel-e2e-expected.yaml]: Might have been a merge conflict snafu. Fixed.

File: cmd/translatesfx/translatesfx/otel.go (hunk 3 above)
Comment (pmcollins) [reply cmd/translatesfx/translatesfx/otel.go:54]: I was thinking the forwarder could be added only if it was present in the SA config. Would it be better to always add it?

**Comment by pmcollins** [comment]
Link checker failure appears to be spurious/unrelated

**Review by rmfitzpatrick — APPROVED** [review]

**Comment by pmcollins** [comment]
Thanks for the review @rmfitzpatrick!

**Pull request merged**